src/main/java/minijava/Cli.java

```diff
@@ -267,10 +277,9 @@ static Parameters parse(String... args) {
       Parameters params = new Parameters();
       try {
         new JCommander(params, args);
-        if (params.mainParameters.size() == 1) {
-          params.file = params.mainParameters.get(0);
-        }
+        params.file = params.mainParameters.get(params.mainParameters.size() - 1);
```

Comment:
`echo` is special in that a test broke.

Really, I'm not motivated to dig into this code any longer, I just want to make it work good enough. The CLI works as is (probably?!) and any more changes should really refactor this more into a style that `jCommander` expects.

The reason why I did this, is so that you can optionally specify `--optimize` and `--verbosity`. There's no other way around this not involving major code changes, in a part of the compiler irrelevant to the actual task.

I don't like what I did there (starting from something battle-tested, but spaghettish and make it even fuglier), but boasting with our CLI code is not something I plan to when this is finished.